erts/emulator/beam/external.c

```diff
@@ -6500,7 +6501,295 @@ Sint transcode_dist_obuf(ErtsDistOutputBuf* ob,
             return 0;
         return reds;
     }
-    
+
+    if ((~dflags & DFLAG_ALTACT_SIG)
+        && ep[0] == SMALL_TUPLE_EXT
+        && (ep[1] == 4 || ep[1] == 5)
+        && ep[2] == SMALL_INTEGER_EXT
+        && ep[3] == DOP_ALTACT_SIG_SEND) {
+        byte *ptr;
+        int dist_op;
+        Uint32 flags;
+        int tuple_size = ep[1];
+        ptr = &ep[4];
+
+        /*
+         * The receiver does not understand alternate action signals...
+         *
+         * This control message has the following layouts, depending on
+         * whether it contains a token or not:
+         * - {DOP_ALTACT_SIG_SEND, Flags, SenderPid, To}
+         * - {DOP_ALTACT_SIG_SEND, Flags, SenderPid, To, Token}
+         * The control message is always followed by a payload term which
+         * is the actual message to deliver.
+         *
+         * The Flags element is an integer which currently have these bit
+         * flags defined in the least significant bits:
+         * - ERTS_DOP_ALTACT_SIG_FLG_PRIO
+         *   This is a priority message
+         * - ERTS_DOP_ALTACT_SIG_FLG_TOKEN
+         *   Control message also contains a token (control message is
+         *   a 5-tuple instead of a 4-tuple).
+         * - ERTS_DOP_ALTACT_SIG_FLG_ALIAS
+         *   An alias message (To is a reference)
+         * - ERTS_DOP_ALTACT_SIG_FLG_NAME
+         *   Send to a registered name (To is an atom)
+         * - ERTS_DOP_ALTACT_SIG_FLG_EXIT
+         *   The signal is an exit signal; otherwise, a message signal
+         *
+         * If neither ERTS_DOP_ALTACT_SIG_FLG_ALIAS nor
+         * ERTS_DOP_ALTACT_SIG_FLG_NAME is set, 'To' is a process
+         * identifier.
+         *
+         * Currently the only alternate action messages supported are:
+         * * Alias messages ('To' is a reference)
+         * * Priority messages ('To' is a reference, atom, or pid)
+         *
+         * The receiving node does not support priority messages. By this we
+         * know that the receiver process have not enabled reception of messages
+         * as priority messages, so we can safely send the message as an
+         * ordinary message.
+         *
+         * If we are sending an alias message, we need to check if it supports
+         * DOP_ALIAS_SEND/DOP_ALIAS_SEND_TT control messages and send it as such
```

Comment:
I've forgotten the details of how the transcode things work, but could we not directly encode as `DOP_ALIAS` when we don't know the capabilities? That way we only priority messages would need to be transcoded and they are very trivial?

Or is the aim to be able to remove `DOP_ALIAS` in the future so we don't want to send any of those if we don't need to?

Reply:
> Or is the aim to be able to remove DOP_ALIAS in the future so we don't want to send any of those if we don't need to?

Yes, but I forgot to add deprecations and scheduling for removal of `DOP_ALIAS`. Fixed now.